### PR TITLE
Bugfix/make better gift

### PR DESCRIPTION
**Pull request by froboy**
## Ticket(s)

Please do not submit a Pull Request without a relevant ticket! If you are creating a new pattern or feature, create an issue describing the need for this feature Github **first** and then link to it below.

**Github Issue**
NA

**Jira Ticket**

- [EWL-4266: Deploy is broken](https://issues.ama-assn.org/browse/EWL-4266)

## Description

This adds the `patch-package` package to enable patching of npm modules. Using this required updating to a much more recent version of `npm`, which also changes how lock files are managed. We'll now be switching from shrinkwrap to npm's native lock files. I've also added a new requirements section to make clear the version dependencies.

## To Test

- Read the readme and update your local to be consistent with the given requirements. Run `npm --version` to make sure you're up to >=5.2 (I'm using v5.5.1).
- Run `npm install` to update all the things. If anything breaks, run `rm -rf node_modules && npm install`
- Run `gulp serve` and make sure things look good
- Run `gulp drupal-deploy` to make sure it works (this should be ok to test, as this was based on develop pretty close to the time of last release).

## Relevant Screenshots/GIFs

Use something like [Skitch](https://evernote.com/skitch/) or [GIPHY Capture](https://giphy.com/apps/giphycapture) to capture images/gifs to demonstrate behaviors.

## Remaining Tasks

When this is merged we'll need to notify everyone to update their NPM version ASAP.

## Additional Notes

Anything more to add?

### CODE REVIEW

**Comment by labbydev** [comment]
> Run npm --version to make sure you're up to >=5.2 (I'm using v5.5.1)

This should be added as a requirement for the style guide in the README, you can't install this without it.

**Comment by labbydev** [comment]
> npm WARN ajv-keywords@2.1.1 requires a peer of ajv@^5.0.0 but none is installed. You must install peer dependencies yourself.
This also seems concerning when running `npm install`

File: README.md [hunk 1]
```diff
@@ -23,10 +23,15 @@ This style guide is a compilation of [atomic components](http://bradfrost.com/bl
...
 - `composer install`
-- `npm install`
+- `sudo npm install -g npm`
 - `sudo npm install -g gulp`
   - This will install gulp globally on your machine. If you don't have privileges, don't want to install globally, or need to manage multiple projects using `gulp`, you can invoke `gulp` via `./node_modules/.bin/gulp serve` instead of directly.
 
+**Version Requirements**
+
+**Node.js** - v7.10.1 ( always < v8.x as this has breaking conflicts)
+**npm** - >= v5.2
```
Comment (labbydev) [inline README.md]: Update these to bullets, they show as one line and it's easy to miss the npm requirement when skimming (or only passively reading). Also you should add node to update npm to the mac setup IMO.

**Comment by froboy** [comment]
@labbydev updated.

**Pull request merged**